include/simdjson/generic/ondemand/value_iterator-inl.h

```diff
@@ -628,15 +626,24 @@ inline void value_iterator::assert_at_next() const noexcept {
   SIMDJSON_ASSUME( _depth > 0 );
 }
 
+simdjson_really_inline void value_iterator::move_at_start() noexcept {
+  _json_iter->_depth = _depth;
+  _json_iter->token.index = _start_position;
+}
 
-simdjson_really_inline void value_iterator::enter_at_container_start() noexcept {
-  _json_iter->_depth = _depth + 1;
+simdjson_really_inline void value_iterator::move_at_container_start() noexcept {
+  _json_iter->_depth = _depth;
   _json_iter->token.index = _start_position + 1;
 }
 
-simdjson_really_inline void value_iterator::move_at_start() noexcept {
-    _json_iter->_depth = _depth;
-  _json_iter->token.index = _start_position;
+simdjson_really_inline bool value_iterator::reset_array() noexcept {
+  move_at_container_start();
```

Comment:
As noted above, I'd suggest we remove move_at_container_start() and replace the calls with:

```c++
_json_iter->reenter_child(_start_position + 1, _json_iter->_depth);
```

This way, we don't have to introduce a concept for "inside the container but not knowing if it is empty yet." There is only "inside a container with a value," and "after the container."

Reply:
See my comment later. I am not sure I trust this replacement to be robust.